internal/statemachine/classic_test.go

```diff
@@ -4531,6 +4542,26 @@ func TestStateMachine_installPackages_checkcmds_failing(t *testing.T) {
 	}
 }
 
+func checkDivert(t *testing.T, fn func() error, divert *func(string, bool) (func() error, func(error) error)) {
+	asserter := helper.Asserter{T: t}
+
+	divertSaved := *divert
+
+	*divert = func(targetDir string, debug bool) (func() error, func(error) error) {
+		return func() error { return fmt.Errorf("divert") }, func(err error) error { return err }
+	}
+	err := fn()
+	asserter.AssertErrContains(err, "divert")
+
+	*divert = func(targetDir string, debug bool) (func() error, func(error) error) {
+		return func() error { return nil }, func(err error) error { return errors.Join(err, fmt.Errorf("undivert")) }
+	}
+	err = fn()
+	asserter.AssertErrContains(err, "undivert")
+
+	*divert = divertSaved
```

Comment:
I suggest putting that in a defer, just after `divertSaved := *divert`, to make sure the function will be restored even if something goes wrong.